plugins/plugin-docker/che-plugin-openshift-client/src/main/java/org/eclipse/che/plugin/openshift/client/OpenShiftConnector.java

```diff
@@ -1428,4 +1432,38 @@ private boolean isTerminalAgentInjected(final Set<String> exposedPorts) {
     private boolean isDevMachine(final Set<String> exposedPorts) {
         return exposedPorts.contains(CHE_WORKSPACE_AGENT_PORT + "/tcp");
     }
+
+    @Override
+    public SystemInfo getSystemInfo() throws IOException {
+        PodList chePods = openShiftClient.pods().inNamespace(this.openShiftCheProjectName)
+                .withLabel(OPENSHIFT_DEPLOYMENT_CONFIG_LABEL, OPENSHIFT_DEPLOYMENT_CONFIG_VALUE).list();
+        if (chePods.getItems().size() > 0) {
+            Pod pod = chePods.getItems().get(0);
+            Node node = openShiftClient.nodes().withName(pod.getSpec().getNodeName()).get();
+            NodeSystemInfo nodeInfo = node.getStatus().getNodeInfo();
+            SystemInfo systemInfo = new SystemInfo();
+            systemInfo.setKernelVersion(nodeInfo.getKernelVersion());
+            systemInfo.setOperatingSystem(nodeInfo.getOperatingSystem());
+            systemInfo.setID(node.getMetadata().getUid());
+            int containers = 0;
+            List<Pod> pods = openShiftClient.pods().inNamespace(this.openShiftCheProjectName).list().getItems();
+            for (Pod p:pods) {
+                containers += p.getSpec().getContainers().size();
```

Comment:
In our current implementation, we are using 1 container/pod, so `containers` should just be equal to the number of pods in the eclipse-che namespace.